x-pack/plugins/security_solution/public/common/components/event_details/osquery_tab.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiNotificationBadge } from '@elastic/eui';
+import React from 'react';
+import styled from 'styled-components';
+import { RESPONSE_ACTION_TYPES } from '../../../../common/detection_engine/rule_response_actions/schemas';
+import { useIsExperimentalFeatureEnabled } from '../../hooks/use_experimental_features';
+import { useKibana } from '../../lib/kibana';
+import type { AlertRawEventData } from './event_details';
+import { EventsViewType } from './event_details';
+import * as i18n from './translations';
+import { useHandleAddToTimeline } from './add_to_timeline_button';
+
+const TabContentWrapper = styled.div`
+  height: 100%;
+  position: relative;
+`;
+
+export const useOsqueryTab = ({
+  rawEventData,
+  id,
+}: {
+  rawEventData?: AlertRawEventData;
+  id: string;
+}) => {
+  const {
+    services: { osquery },
+  } = useKibana();
+  const handleAddToTimeline = useHandleAddToTimeline();
+  const responseActionsEnabled = useIsExperimentalFeatureEnabled('responseActionsEnabled');
+  if (!osquery || !rawEventData || !responseActionsEnabled) {
+    return;
+  }
+
+  const { OsqueryResults } = osquery;
+  const parameters = rawEventData.fields['kibana.alert.rule.parameters'];
```

Comment:
would be easier to work with everything in rawEventData if you used the helper expandDottedObject https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/common/utils/alerts.ts#L82

Reply:
I'll be creating a small PR later today with a small fix, do you mind If I attach this change there too? :) Thanks!

Reply:
And yessss, this is much easier, thank you :)